aria_data_loaders/aria_data_utils/aria_live_streamer.py

```diff
@@ -0,0 +1,598 @@
+            detect_objects (bool, optional): Whether to detect objects. Defaults to False.
+            object_label (str, optional): Object label to detect. Defaults to "milk bottle".
+
+        Publishes:
+            - static tf: ROS TransformStamped message for marker frame wrt ariaWorld
+            - /aria_pose_of_interest: ROS PoseStamped message for ariaWorld_T_cpf at pose of interest i.e. pose where it detected the object
+            - /aria_pose_of_interest_marker: ROS Marker message for ariaWorld_T_cpf at pose of interest i.e. pose where it detected the object
+
+        Returns:
+            Tuple[np.ndarray, dict]: Tuple of:
+                - viz_img (np.ndarray): Image with april tag and object in hand detections
+                - outputs (dict): Dictionary of outputs from the april tag detector
+        """
+        # Initialize camera_T_marker to None & object_scores to empty dict for current image frame
+        camera_T_marker = None
+        object_scores = {}
+
+        viz_img = frame.get("image")
+        if detect_qr:
+            (viz_img, camera_T_marker) = self.april_tag_detector.process_frame(img_frame=frame.get("raw_image"))  # type: ignore
+            # Rotate current image frame
+            viz_img = self._rotate_img(img=viz_img)
+
+        if camera_T_marker is not None:
+            device_T_marker = self.device_T_camera * camera_T_marker
+            ariaWorld_T_device = frame.get("device_pose")
+
+            # Frame: world = ariaWorld
+            # Frame: a = device (aria's device / left slam camera frame)
+            # Frame: b = marker / qr code
+            (
+                self.marker_positions_list,
+                self.marker_quaternion_list,
+                self.avg_ariaWorld_T_marker,
+            ) = get_running_avg_world_T_b(
+                current_avg_world_T_b=self.avg_ariaWorld_T_marker,
+                world_T_b_position_list=self.marker_positions_list,
+                world_T_b_quaternion_list=self.marker_quaternion_list,
+                world_T_a=ariaWorld_T_device,
+                a_T_b=device_T_marker,
+            )
+
+            if self.avg_ariaWorld_T_marker is not None:
+                avg_marker_T_ariaWorld = self.avg_ariaWorld_T_marker.inverse()
+
+                # Publish marker pose in ariaWorld frame
+                self.static_tf_broadcaster.sendTransform(
+                    sophus_SE3_to_ros_TransformStamped(
+                        sp_se3=avg_marker_T_ariaWorld,
+                        parent_frame=rf.MARKER,
+                        child_frame=rf.ARIA_WORLD,
+                    )
+                )
+
+            viz_img, outputs = self.april_tag_detector.get_outputs(
+                img_frame=viz_img,
+                outputs=outputs,
+                device_T_marker=device_T_marker,
+                timestamp=frame.get("timestamp"),
+                img_metadata=None,
+            )
+
+        if detect_objects:
```

Comment:
is it either detect_object or detect_qr? What happens if both are false?

Reply:
We had both ON while running P1 demo as we wanted to detect QR when it pops up and also detect objects throughout the episode. If both are false, then function just returns the input image as is and an empty dictionary.
We coded it like this as it made easy to test one or the other while debugging.